packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1451,6 +1461,25 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    // Track how many times the container tries to reconnect with pending messages.
+    // This happens when the connection state is changed and we reset the counter
+    // when we are able to process a local op.
+    // If this counter reaches a max, it's a good indicator that the container
+    // is not making progress and it is stuck in a retry loop.
+    private shouldContinueReconnecting(): boolean {
+        if (this.maxConsecutiveReconnects < 0) {
+            // Feature disabled, we assume we are always making progress
+            return true;
+        }
+
+        return !this.pendingStateManager.hasPendingMessages()
```

Comment:
I'd think that if  this.pendingStateManager.hasPendingMessages() === false, then you want to reset this.consecutiveReconnects rigth here.
unlikely, but imagine the situation that we failed to send ops 14 times, but them on connection DDSs decided that rebasing of ops resulted in noop, i.e. no local changes. It would be great not to fail container on next disconnect :)